web/src/containers/ApdPreviousActivityTable.js

```diff
@@ -0,0 +1,162 @@
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+
+import { formatMoney } from '../util/formats';
+import { DollarInput } from '../components/Inputs';
+import { updateApd } from '../actions/apd';
+
+const ApdPreviousActivityTable = ({
+  previousActivityExpenses,
+  updateApd: dispatchUpdateApd
+}) => {
+  const programs = { hit: 'HIT', hie: 'HIE', hitAndHie: 'HIT + HIE' };
+  const colors = ['aqua', 'blue', 'navy'];
+  const years = Object.keys(previousActivityExpenses);
+
+  const handleChange = (year, program, type) => e => {
+    const update = {
+      previousActivityExpenses: {
+        [year]: { [program]: { [type]: e.target.value } }
+      }
+    };
+    dispatchUpdateApd(update);
+  };
+
+  const borderClass = i => `border-left border-${i < 0 ? 'gray' : colors[i]}`;
+
+  return (
+    <table
+      className={`h6 table-fixed table-condensed border-right border-${
+        colors[2]
+      }`}
+      style={{ minWidth: 1200 }}
+    >
+      <thead>
+        <tr>
+          <th />
+          {Object.values(programs).map((name, i) => (
+            <th
+              key={name}
+              colSpan={4 + (i === 2 ? 2 : 0)}
+              className={`bg-${colors[i]} white center border border-${
+                colors[i]
+              }`}
+            >
+              {name}
+            </th>
+          ))}
+        </tr>
+        <tr>
+          <th className="border-none" />
+          {Object.values(programs).map((name, i) => (
+            <Fragment key={name}>
+              <th colSpan="2" className={borderClass(i)}>
+                Federal share<br />90% FFP
+              </th>
+              <th colSpan="2" className={borderClass(i)}>
+                State share<br />10% FFP
+              </th>
+            </Fragment>
+          ))}
+          <th colSpan="2" className={`border-right ${borderClass(2)}`}>
+            Federal + state<br />Grand total computable
+          </th>
+        </tr>
+        <tr>
+          <th />
+          {Object.values(programs).map((name, i) => (
+            <Fragment key={name}>
+              <th className={`bg-${colors[i]}-light ${borderClass(i)}`}>
```

Comment:
these dynamic classes (which I've done a bunch of) feel quite messy to me; fine for now but I'm making a mental note - and actual note here :) - to revisit and clean up.